### DISCUSSION TITLE
Git sparse command usage and risks

### DISCUSSION THREAD

**Discussion by ashwinmalaviya** [community/community#29935] [Other Feature Feedback, Questions, & Ideas]
In our project we deal with very very large code organization which is hosted in github. at a time checking out this large code base takes lot of time and it becomes really difficult to deal with this kind of monorepo. 
however recently i came across Git sparse checkout command which is exactly we need and will solve our issue.
however when i read the description of this command there is waring written in capital letters which says
"THIS COMMAND IS EXPERIMENTAL. ITS BEHAVIOR, AND THE BEHAVIOR OF OTHER COMMANDS IN THE PRESENCE OF SPARSE-CHECKOUTS, WILL LIKELY CHANGE IN THE FUTURE."
i am not able to understand this line. does it mean that use of this command may change in future and we will not be able to use sparse checkout in future? 
does any one know the risk associated with this command? 
also is there any alternative available to this command which we can use but provides the similar experience where we do not need to checkout full repo but just the few folders which they require.

thanks in advance for providing your valuable inputs.

**Comment by airtower-luna** [comment]
> does it mean that use of this command may change in future and we will not be able to use sparse checkout in future?

Either that, or it won't work in the same way. And it may be buggy. You should definitely check the changelog when updating Git.

In general, "experimental" in the description of software means that it's not complete yet, let alone well tested. Less politely the warning means: You can try this out if you want, but be prepared to face bugs, and don't come crying if it breaks things or we decide to scrap it after all. :sweat_smile: 

On the plus side, if you are prepared to deal with bugs, report them, test patches, etc. maybe you can help development along. :slightly_smiling_face:

**Reply by ashwinmalaviya** [reply]
thanks for your answer. but it seems gilab supports this command with some filter mechanism 
https://docs.gitlab.com/ee/topics/git/partial_clone.html. 
do you have any idea if it can be implemented in github?

**Reply by airtower-luna** [reply]
It should (as in, I haven't tried, but there's no reason it shouldn't) work with GitHub the same way. The warning is a general one, it applies to *any* use of Git with sparse checkout, not a specific hoster. :wink:
